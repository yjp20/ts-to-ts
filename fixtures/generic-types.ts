/** @model */
interface Container<T> {
  value: T;
  metadata?: Record<string, unknown>;
}

/** @model */
type StringContainer = Container<string>;
/** @model */
type NumberList = Array<number>;
/** @model */
type Dictionary = Record<string, any>;
