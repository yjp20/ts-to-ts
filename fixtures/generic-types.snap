
<<<<<<<<<<< typescript <<<<<<<<<<<<
/** @model */
interface Container<T> {
  value: T;
  metadata?: Record<string, unknown>;
}

/** @model */
type StringContainer = Container<string>;
/** @model */
type NumberList = Array<number>;
/** @model */
type Dictionary = Record<string, any>;
===================================
// generic-types.ts
model StringContainer<T> {
  value: string,
  metadata?: Record<never, never>
};

alias NumberList<T> = float64[];

alias Dictionary = Record<never, never>;

model Container<T> {
  value: T,
  metadata?: Record<never, never>
};
>>>>>>>>>>> typespec >>>>>>>>>>>>>>